src/logging.js

```diff
@@ -1,57 +1,96 @@
+    level = chalk[TERM_COLORS[level]](level);
+  }
+
+  return options.timestamp() + level + ' ' + (options.message ? options.message : '') +
+    (options.meta && Object.keys(options.meta).length ? '\n\t' + JSON.stringify(options.meta) : '' );
+}
 
 function init (loggingConfig) {
   const transports = [];
   if (loggingConfig == null) {
     loggingConfig = { };
   }
   if (loggingConfig.level == null) {
-    loggingConfig.level = "info";
+    loggingConfig.level = DEFAULT_LOG_LEVEL;
+  }
+
+  if (loggingConfig.rotate == null) {
+    loggingConfig.rotate = {};
   }
-  transports.push(new (winston.transports.Console)({
+
+  const funcTimestamp = () => new Date().toISOString().replace(/[TZ]/g, ' ');
+  const showConsole = loggingConfig.console !== false;
+  const fileCfg = {
     json: false,
-    name: "console",
-    timestamp: () => new Date().toISOString().replace(/[TZ]/g, ' '),
-    formatter: function (options) {
-      return options.timestamp() + options.level + ' ' + (options.message ? options.message : '') +
-        (options.meta && Object.keys(options.meta).length ? '\n\t' + JSON.stringify(options.meta) : '' );
-    },
+    name: "file",
+    filename: loggingConfig.file,
+    timestamp: funcTimestamp,
+    formatter: winstonFormatter,
     level: loggingConfig.level,
-  }));
+    maxsize: loggingConfig.rotate.size,
+    maxFiles: loggingConfig.rotate.count,
+    zippedArchive: loggingConfig.compress,
+  }
+  if (loggingConfig.file ) {
+    if (loggingConfig.rotate.daily !== true) {
+      transports.push(new (winston.transports.File)(fileCfg));
+    } else {
+      fileCfg.datePattern = loggingConfig.rotate.datePattern;
+      transports.push(new (winston.transports.DailyRotateFile)(fileCfg));
+    }
+  }
+
 
-  if (loggingConfig.file) {
-    const logrotateConfig = {
-      file: loggingConfig.file,
+  if (showConsole) {
+    transports.push(new (winston.transports.Console)({
       json: false,
-      size: loggingConfig.size,
-      keep: loggingConfig.count,
-      compress: loggingConfig.compress,
-      timestamp: () => new Date().toISOString().replace(/[TZ]/g, ' '),
-      formatter: function (options) {
-        return options.timestamp() + options.level + ' ' + (options.message ? options.message : '') +
-          (options.meta && Object.keys(options.meta).length ? '\n\t' + JSON.stringify(options.meta) : '' );
-      },
+      name: "console",
+      timestamp: funcTimestamp,
+      formatter: winstonFormatter,
       level: loggingConfig.level,
-    };
-    transports.push(new Rotate(logrotateConfig));
+      colorize: true,
+      prettyPrint: true,
+    }));
   }
 
 
   log = new winston.Logger({
     transports: transports,
+    levels: {
+      error: 0,
+      warn: 1,
+      info: 2,
+      verbose: 3,
+      silly: 4,
+    }
   });
+  if (loggingConfig.rotate.size && loggingConfig.rotate.daily) {
+    handle("warn", ["You have enabled both 'size' and 'daily' in your logger config. Size will be ignored."]);
```

Comment:
I think it'd be best to just to log an error and die if both are set. It doesn't make sense to have both configured and might be a source of confusion.